kbs/src/plugins/implementations/nebula_ca.rs

```diff
@@ -0,0 +1,537 @@
+        // store state.
+        match sub_path {
+            // Create credential for the provided parameters.
+            // The credential directory (and its files) is auto-deleted after the Credential is returned.
+            "credential" => {
+                let params = NebulaCredentialParams::try_from(query)?;
+
+                let credential_dir = tempdir_in(self.work_dir.as_path())?;
+                let node_key: PathBuf = credential_dir.path().to_owned().join("node.key");
+                let node_crt: PathBuf = credential_dir.path().to_owned().join("node.crt");
+
+                let credential = self
+                    .create_credential(node_key.as_path(), node_crt.as_path(), &params)
+                    .await?;
+
+                Ok(serde_json::to_vec(&credential)?)
+            }
+            _ => Err(anyhow!("{} not supported", sub_path))?,
+        }
+    }
+
+    async fn validate_auth(
+        &self,
+        _body: &[u8],
+        _query: &str,
+        _path: &str,
+        _method: &Method,
+    ) -> Result<bool> {
+        Ok(false)
+    }
+
+    async fn encrypted(
+        &self,
+        _body: &[u8],
+        _query: &str,
+        _path: &str,
+        _method: &Method,
+    ) -> Result<bool> {
+        Ok(true)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use rstest::rstest;
+    use std::ffi::OsString;
+
+    use super::NebulaCredentialParams;
+
+    #[rstest]
+    #[case(
+        "name=pod1&ip=1.2.3.4/21",
+        Some(vec![
+            "-name".into(),
+            "pod1".into(),
+            "-ip".into(),
+            "1.2.3.4/21".into()
+        ])
+    )]
+    #[case(
+        "name=pod1&ip=1.2.3.4",
+        Some(vec![
+            "-name".into(),
+            "pod1".into(),
+            "-ip".into(),
+            "1.2.3.4".into()
+        ])
+    )]
+    #[case(
+        "name=pod1&ip=1.2",
+        Some(vec![
+            "-name".into(),
+            "pod1".into(),
+            "-ip".into(),
+            "1.2".into()
+        ])
+    )]
+    #[case("name=pod1", None)]
+    #[case(
+        "name=pod1&ip=1.2.3.4/21&duration=8760h10m10s",
+        Some(vec![
+            "-name".into(),
+            "pod1".into(),
+            "-ip".into(),
+            "1.2.3.4/21".into(),
+            "-duration".into(),
+            "8760h10m10s".into(),
+        ])
+    )]
+    #[case(
+        "name=pod1&ip=1.2.3.4/21&groups=server,ssh",
+        Some(vec![
+            "-name".into(),
+            "pod1".into(),
+            "-ip".into(),
+            "1.2.3.4/21".into(),
+            "-groups".into(),
+            "server,ssh".into(),
+        ])
+    )]
+    #[case(
+        "name=pod1&ip=1.2.3.4/21&subnets=1.2.3.5/21,1.2.3.6/21",
+        Some(vec![
+            "-name".into(),
+            "pod1".into(),
+            "-ip".into(),
+            "1.2.3.4/21".into(),
+            "-subnets".into(),
+            "1.2.3.5/21,1.2.3.6/21".into(),
+        ])
+    )]
+
```

Comment:
```suggestion
```

Let 's delete the empty line

Reply:
Actually, I will need to keep it.
`make lint` fails if an empty line is not found at the end of the file.